src/Microsoft.Extensions.Configuration.Abstractions/ConfigurationPath.cs

```diff
@@ -42,6 +60,11 @@ public static string GetSectionKey(string path)
             return lastDelimiterIndex == -1 ? path : path.Substring(lastDelimiterIndex + 1);
         }
 
+        /// <summary>
+        /// Extracts the parent path from a path.
+        /// </summary>
+        /// <param name="path">The path.</param>
+        /// <returns>The parent part of the path.</returns>
```

Comment:
The original path minus the last individual key found in it. Null if the original path corresponds to a top level node.